src/Groups/homomorphisms.jl

```diff
@@ -616,6 +616,26 @@ function isomorphism(::Type{GrpAbFinGen}, A::GrpAbFinGen)
    end::AbstractAlgebra.Generic.IdentityMap{GrpAbFinGen}
 end
 
+# We need not find independent generators in order to create
+# a presentation of a fin. gen. abelian group.
+function isomorphism(::Type{FPGroup}, A::GrpAbFinGen)
+   # Known isomorphisms are cached in the attribute `:isomorphisms`.
+   isos = get_attribute!(Dict{Type, Any}, A, :isomorphisms)
+   return get!(isos, FPGroup) do
+      G = free_group(ngens(A))
+      R = rels(A)
+      s = vcat(elem_type(G)[i*j*inv(i)*inv(j) for i = gens(G) for j = gens(G) if i != j],
+           elem_type(G)[reduce(*, [gen(G, i)^R[j,i] for i=1:ngens(A) if !iszero(R[j,i])], init = one(G)) for j=1:nrows(R)])
+      F, mF = quo(G, s)
+      @assert is_finite(A) == is_finite(F)
```

Comment:
maybe turn them into hassert? So that they are not always run? Is IsFinite not potentially a desaster for fp-groups - or is it clear that the presentations here are trivial?

Reply:
@fieker I had kept your assertion that `order(A) == order(F)` holds, but this statement can be evaluated only if the two groups are finite, thus I have restricted the check to finite groups, and then added the additional assertion for checking `is_finite(A) == is_finite(F)`.
This additional assertion is not critical here, already because even the group orders get computed when the original assertion gets executed. However, we can of course remove the assertions.

Would it make sense to document `@hassert` in order to advertise it for Oscar code?